Indexer ready for integration testing

mikebaumann opened the pull request:
Note that the indexer unit test now expects a local Elasticsearch instance to be running, and fails with a message to that effect if it is not. For developers this may be fine, yet we need to consider the effect on the Travis builds.

kislyuk commented:
please fix the tests, it appears moto needs to be added to requirements-dev.txt

kislyuk commented on tests/expected_index_document.json:
```diff
@@ -0,0 +1,166 @@
+            "uuid": "5aca931f-9c88-4483-ba46-685b70c08a73"
+        }
+    }
+}
```
For all of these data bundle files, I suggest that you pull them from the submodule instead of copying them here.

mikebaumann replied:
Thanks, I have changed the tests to use the files data-bundle-examples submodule.

mikebaumann commented:
Thanks for your review! I added moto to requirements-dev.txt and worked with you to add Elasticsearch to the Travis build. As noted above, I also changed the index text to use data from the data-bundle-examples submodule, including dynamically creating and removing files in the bundle that are not to be indexed, for test purposes. The manifest (including the non-indexed files) is included in the index document (as per the spec) but the content of the non-indexed files are not included in the index document. The combination of both indexed and non-indexed files in a bundle is tested.
